### DISCUSSION TITLE
[Extension] : How to add new menu item to query result right click context menu?

### DISCUSSION THREAD

**Discussion by osmant** [microsoft/azuredatastudio#24628] [Q&A] [answered]
Hi all,

I'm developing an extension and need to add a menu item to the context menu in the Result grid of Azure Data Studio.

Challenge:

Adding an item to the right-click context menu of the Result grid, not the editor.
Questions:

How can I add items to the Result grid's context menu?

Regards

**Comment by Charles-Gagnon** [comment, marked as answer]
Unfortunately that is not possible without changes to the core code. The result grid actions are hardcoded with a set of actions.

https://github.com/microsoft/azuredatastudio/blob/25cdb558a707b55ada24b58489b9ad387dbde42d/src/sql/workbench/contrib/query/browser/gridPanel.ts#L1020 here's the code if you're interested.

If this is a feature you'd like to see please open a feature request issue and we can take a look!

**Reply by osmant** [reply]
Thanks for the quick reply. I will open feature request issue with the use case I have in my mind.
